CONTRIBUTING.md

```diff
@@ -70,21 +91,58 @@ Setting Up Your Environment
   itself, the "fast forward" pull will update your local branch without adding
   a merge commit.
 
-7. If you need to update a local branch you are working on, you may "pull" the
-  upstream branch or rebase:
+  (If you have made any local conflicting changes to the `develop` branch itself,
+  then you're doing it wrong.  The use of the `--ff-only` option will detect this
+  for you and refuse to work.  As per point (5) above, all your changes should
+  be happening in a different branch other than `develop`.)
+
+7. If the upstream develop branch changes while you are working on your branch,
+  so you need to update your branch, you may (and probably should) merge the
+  upstream `develop` branch directly into it or rebase:
   ```
+  git checkout whatever-the-branch-i-am-working-on-is-called
   git pull upstream develop
   ```
 
   or:
   ```
+  git checkout whatever-the-branch-i-am-working-on-is-called
   git rebase upstream develop
   ```
 
   See the notes above regarding `rebase`
+
+  This makes life easier for the other developer who has to merge your pull
+  request into the upstream `develop`.  You are better suited to understanding
+  how to deal with the potential merge conflicts that could come up between
+  your branch and the upstream `develop` changes than somebody else is.
+
 8. You may push your branch with edits to your `origin` on github, and submit a
   pull request to KSP-KOS/KOS `develop` for review to be included.
 
+####Nobody merges their own PR's.
```

Comment:
I don't really see moving it as an improvement, but I also don't care, so sure I'll move it.